applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaTypes.h

```diff
@@ -634,6 +638,163 @@ inline const char* CudaRigid3fTypes::Name()
     return "CudaRigid3f";
 }
 
+//=============================================================================
+// 2D Rigids
+//=============================================================================
+
+template<typename real>
+class CudaRigidTypes<2, real>
+{
+public:
+    typedef real Real;
+    typedef typename sofa::defaulttype::StdRigidTypes<2,Real>::Vec2 Vec2;
+
+    typedef sofa::defaulttype::RigidDeriv<2,real> Deriv;
+    typedef sofa::defaulttype::RigidCoord<2,real> Coord;
+    typedef Real AngularVector;
+
+    enum { spatial_dimensions = Coord::spatial_dimensions };
+    enum { coord_total_size = Coord::total_size };
+    enum { deriv_total_size = Deriv::total_size };
+
+    typedef typename Coord::Pos CPos;
+    typedef typename Coord::Rot CRot;
+    static const CPos& getCPos(const Coord& c) { return c.getCenter(); }
+    static void setCPos(Coord& c, const CPos& v) { c.getCenter() = v; }
+    static const CRot& getCRot(const Coord& c) { return c.getOrientation(); }
+    static void setCRot(Coord& c, const CRot& v) { c.getOrientation() = v; }
+
+    typedef typename sofa::defaulttype::StdRigidTypes<2,Real>::DPos DPos;
+    typedef real DRot;
+    static const DPos& getDPos(const Deriv& d) { return getVCenter(d); }
+    static void setDPos(Deriv& d, const DPos& v) { getVCenter(d) = v; }
+    static const DRot& getDRot(const Deriv& d) { return getVOrientation(d); }
+    static void setDRot(Deriv& d, const DRot& v) { getVOrientation(d) = v; }
+
+    static const char* Name();
+
+    typedef CudaVector<Coord> VecCoord;
+    typedef CudaVector<Deriv> VecDeriv;
+    typedef CudaVector<Real> VecReal;
+
+    typedef defaulttype::MapMapSparseMatrix<Deriv> MatrixDeriv;
+
+    template<typename T>
+    static void set(Coord& c, T x, T y, T)
```

Comment:
I never used rigid2Types so I'm not sure of the API but it would make sens to me to add ```set```, ```get``` and ```add``` methods below with a version with only (T& x, T& y , without the z variable which is then ignored.

Reply:
Sorry, I'm not familiar with GitHub, and I accidentally clicked the **request for reveiw** again.....😢